### ISSUE TITLE
backend support loading individual Stokes images for visualization with the Stokes analysis widget

### ISSUE THREAD

**Issue by kswang1029** [CARTAvis/carta-backend#611]
With the current implementation, the Stokes analysis widget is functional only if the loaded image is 4D with Stokes IQU or IQUV.

However, it is common that Stokes images are stored as separate files. We will need to enhance the Stokes analysis widget so that it can still display polarization info properly from separate Stokes image files. 

related to https://github.com/CARTAvis/carta-frontend/issues/1153

**Comment by markccchiang** [comment]
I think the backend can apply `casacore::ExtendImage` and `casacore::ImageConcat` to concatenate individual stokes images as one without copying the data. That way could much simplify the code changes. Just the backend needs to know which images with respect to their stokes types to combine. The frontend has to change as well. @kswang1029 maybe you can provide the idea of how the UI panel looks like for users to set the combination of the images.

**Comment by veggiesaurus** [comment]
I guess one of the things we need to decide on is whether this combination happens when _loading_ an image, or after images have been loaded

**Comment by markccchiang** [comment]
I have created branches `mark/concatenate_images` for the [backend](https://github.com/CARTAvis/carta-backend/tree/mark/611_concatenate_images) and [protobuf](https://github.com/CARTAvis/carta-protobuf/tree/mark/concatenate_images). A new command `CONCAT_STOKES_FILES` with a new message `ConcatStokesFiles` tells the backend to concatenate stokes files as one. And the backend responses to the command `CONCAT_STOKES_FILES_ACK` with the message `ConcatStokesFilesAck`. If concatenation success, an `OpenFileAck` will be attached in `ConcatStokesFilesAck` and tell the frontend that the backend has opened a new concatenate image. The definition of new protobuf messages are shown below:

![截圖 2020-11-09 下午1 19 05](https://user-images.githubusercontent.com/5379602/98502923-4bf54e00-228e-11eb-8262-b9814206cf8b.png)

where

![截圖 2020-11-09 下午1 21 33](https://user-images.githubusercontent.com/5379602/98503018-83fc9100-228e-11eb-940c-5db45af8bdf5.png)

The backend requires images to be concatenated should have the same shape and same directional and spectral axes indices. The image type (CASA or FITS) should be the same too. The image's full name and the assignment of the stokes type can not be repeated. I have tested my code, so this concatenation function should work. Just it needs further tests with the frontend when it is done.
